.github/workflows/lint.yaml

```diff
@@ -0,0 +1,49 @@
+name: Linting
+
+on:
+  pull_request: {}
+  push:
+    branches:
+        - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  pull-requests: write # for comments
+
+jobs:
+  compatibility:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout tree
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        fetch-tags: true
+
+    - name: Set-up OCaml
+      uses: ocaml/setup-ocaml@v2
```

Comment:
Nice! Coincidence, I started to use this ocaml/setup-ocaml@v2 also yesterday! (for semgrep-server).
It seems to work pretty well.